### PR TITLE
Javascript javascript1 week4

### PR DESCRIPTION
**Pull request by RemyaManoharan**

### CODE REVIEW

File: javascript/javascript1/week4/voice_assistance.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
+let fName = "";
+const todos = [];
```
Comment (naresh699) [inline javascript/javascript1/week4/voice_assistance.js:2]: no wrong code. just read about 'mutate and non-mutatable code in js'. check why we can change values in array even it is 'const'.

File: javascript/javascript1/week4/voice_assistance.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      return `${todo} is not on your todo list`;
+    }
+  }
+  // select items from todos
+  if (action === "what" && words.includes("todo")) {
+    const todoCount = todos.length;
+    if (todoCount === 0) {
+      return "You have no todos";
+    } else {
+      let todoList = "You have " + todoCount + " todos";
```
Comment (naresh699) [inline javascript/javascript1/week4/voice_assistance.js]: try to make consistent coding language. In line no 40, a template string is used to declare values. here is again legacy method

File: javascript/javascript1/week4/voice_assistance.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+  if (action === "hello" && words.includes("name")) {
+    fName = words.slice(words.indexOf("is") + 1, words.length).join(" ");
+    return `Nice to meet you ${fName}`;
+  }
+
+  if (action === "what" && words.includes("name")) {
+    if (fName) {
+      return `Your name is ${fName}`;
+    } else return `I don't know your name`;
```
Comment (naresh699) [inline javascript/javascript1/week4/voice_assistance.js]: if you elemeneting { from else statement you can remove from if also, even better use 'tertiary js condition'

File: javascript/javascript1/week4/voice_assistance.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
+let fName = "";
+const todos = [];
+let timer;
+function getReply(command) {
```
Comment (naresh699) [inline javascript/javascript1/week4/voice_assistance.js:67]: from my perspective, the function is too long with the if else condition. try to split the function. maybe another function that can accept 'action and include the word' as a parameter.

File: javascript/javascript1/week4/voice_assistance.js (hunk 1 above)
Comment (RemyaManoharan) [reply javascript/javascript1/week4/voice_assistance.js:2]: ok

**Review by naresh-pandora — APPROVED** [review]

**Comment by RemyaManoharan** [comment]
Thanks for the review..valuable feedbacks..

**Pull request merged**